include/caf/leds.rst

```diff
@@ -69,54 +68,74 @@ To define PWM ports used by the application, complete the following steps:
 
-	* ``led_pins`` - contains PWM port and pin mapping for each LED used in the application.
+	* Enabling or create relevant nodes in a board-specific :file:`dts` file.
+	* Enabling or create relevant nodes in :file:`dts.overlay` file, which are to be merged to configuration.
 
-	The size of the array is determined by the :option:`CONFIG_CAF_LEDS_COUNT` and :option:`CONFIG_CAF_LEDS_COLOR_COUNT` Kconfig options.
 	The following code snippets show several examples of how the file contents could look:
 
 	* Example 1:
 
-	.. code-block:: c
+	.. code-block:: none
 
-		static const size_t led_pins[CONFIG_CAF_LEDS_COUNT]
-					    [CONFIG_CAF_LEDS_COLOR_COUNT] = {
-			{
-				DT_PROP(DT_NODELABEL(pwm0), ch0_pin)
-			}
+		&pwm_led0 {
+			status = "okay";
+			pwms = <&pwm0 15>;
+			label = "LED0";
 		};
 
 	* Example 2:
 
-	.. code-block:: c
-
-		static const size_t led_pins[CONFIG_CAF_LEDS_COUNT]
-					    [CONFIG_CAF_LEDS_COLOR_COUNT] = {
-			{
-				DT_PROP(DT_NODELABEL(pwm0), ch0_pin),
-				DT_PROP(DT_NODELABEL(pwm0), ch1_pin),
-				DT_PROP(DT_NODELABEL(pwm0), ch2_pin)
-			},
-			{
-				DT_PROP(DT_NODELABEL(pwm1), ch0_pin),
-				DT_PROP(DT_NODELABEL(pwm1), ch1_pin),
-				DT_PROP(DT_NODELABEL(pwm1), ch2_pin)
-			}
+	.. code-block:: none
+
+		pwmleds0 {
+			compatible = "pwm-leds";
+			status = "okay";
+
+			pwm_led0: led_pwm_0 {
+				status = "okay";
+				pwms = <&pwm0 1>;
+				label = "LED0 red";
+			};
+
+			pwm_led1: led_pwm_1 {
+				status = "okay";
+				pwms = <&pwm0 2>;
+				label = "LED0 green";
+			};
+
+			pwm_led2: led_pwm_2 {
+				status = "okay";
+				pwms = <&pwm0 3>;
+				label = "LED0 blue";
+			};
 		};
 
-3. Specify location of the configuration file with the :option:`CONFIG_CAF_LEDS_DEF_PATH` Kconfig option.
+		pwmleds1 {
+			compatible = "pwm-leds";
+			status = "okay";
+
+			pwm_led3: led_pwm_3 {
+				status = "okay";
+				pwms = <&pwm1 4>;
+				label = "LED1";
+			};
+		};
 
-	.. note::
-		The configuration file should be included only by the configured module.
-		Do not include the configuration file in other source files.
+	Make sure to configure all PWM ports and channels that are used by the application.
+	The module assumes that a single pwmleds node is a separate and complete logical LED.
+	The module expects that a single pwmleds node will hold configuration of HW LEDs responsible for reproducing all required color channels.
+	The number of HW LEDs configured to reproduce collor channels can be either one or three.
+	In case one HW LED is used the module will convert tri-channel color to a single value of brightness and pass it to this HW LED.
+
+	For more help, see :ref:`zephyr:dt-guide`.
 
 .. warning::
-	   For the PWM ports to be configured correctly, both the configuration file and the :file:`dts` file must match.
+	   For the LEDs to be configured correctly, pin numbers set for pwm and pwmleds nodes in :file:`dts` file must match.
 
 Implementation details
```

Comment:
How do the changes to the configuration and the switch to pwmleds affect Implementation details?

Reply:
In the end there is little difference. We still control each separate HW LED brightness by controlling the pulse width. Before the change we used PWM driver directly and now we do it via PWM LED driver.

This section will change after we will add GPIO LEDs as there we should abandon referring to PWM altogether and use only brightness.
I don't think there is any point in doing it now however.